### PR TITLE
Misc Typescript Fixes

### PR DESCRIPTION
**Pull request by lbwexler**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [X] Caught up with `develop` branch as of last change.
- [X] Added CHANGELOG entry, or determined not required.
- [X] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [X] Updated doc comments / prop-types, or determined not required.
- [X] Reviewed and tested on Mobile, or determined not required.
- [X] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: cmp/grid/filter/GridFilterFieldSpec.ts [hunk 1]
```diff
@@ -42,7 +41,7 @@ export interface GridFilterFieldSpecConfig extends BaseFilterFieldSpecConfig {
 export class GridFilterFieldSpec extends BaseFilterFieldSpec {
     filterModel: GridFilterModel;
     renderer: ColumnRenderer;
-    inputProps: PlainObject;
+    inputProps: HoistInputProps;
```
Comment (amcclain) [inline cmp/grid/filter/GridFilterFieldSpec.ts:44]: See #3510 - @saba-mo and I just ran into that issue last week, would be great if we could address as part of this PR, taking this one step further to support a broader union type here and on the `GridFilterFieldSpecConfig` interface

File: cmp/zoneGrid/impl/ZoneGridPersistenceModel.ts [hunk 2]
```diff
@@ -22,7 +23,14 @@ export class ZoneGridPersistenceModel extends HoistModel {
...
     @observable.ref
-    state: PlainObject;
+    state: {
+        columns?: Partial<ColumnState>[];
+        sortBy?: GridSorterLike;
+        groupBy?: Some<string>;
+        version?: number;
+        autosize?: any;
+        mappings?: Record<Zone, Some<string | ZoneMapping>>;
+    };
```
Comment (amcclain) [inline cmp/zoneGrid/impl/ZoneGridPersistenceModel.ts:33]: I don't believe we save `columns` or `autosize` in this state block - N/A / replaced by `mappings`

File: data/Store.ts [hunk 3]
```diff
@@ -219,7 +219,6 @@ export class Store extends HoistBase {
     _filtered: RecordSet;
 
     private _dataDefaults = null;
-    private _created = Date.now();
```
Comment (amcclain) [inline data/Store.ts:222]: Turns out we read and show this in the instances grid within Hoist Inspector. There are also `_created` member vars on HoistModel and HoistService, although you can tell how essential they are by the fact that the HoistModel one was broken until just now (https://github.com/xh/hoist-react/commit/b0418d03e4982d8d6a6d1b6f9125dfc3f88d6886)

I don't feel strongly about it, but if we keep the timestamp we should not mark as private, obvs, and if we don't think it's useful we might as well remove it from the other two base classes and the little grid in Inspector

**Review by amcclain — COMMENTED** [review]
Nice set of fixes/cleanups/improvements all around 👍

File: cmp/zoneGrid/impl/ZoneGridPersistenceModel.ts (hunk 2 above)
Comment (lbwexler) [reply cmp/zoneGrid/impl/ZoneGridPersistenceModel.ts:33]: good catch

File: data/Store.ts (hunk 3 above)
Comment (lbwexler) [reply data/Store.ts:222]: I'll revert!  Was thrown off by the mistaken use of private.

File: cmp/grid/filter/GridFilterFieldSpec.ts (hunk 1 above)
Comment (lbwexler) [reply cmp/grid/filter/GridFilterFieldSpec.ts]: will hold off on this change then, and take the other simpler changes tin this PR

**Pull request merged**